pkg/scheduler/core/generic_scheduler.go

```diff
@@ -413,24 +422,20 @@ func (g *genericScheduler) findNodesThatFit(pod *v1.Pod, nodes []*v1.Node) ([]*v
 			}
 			if fits {
 				filtered[atomic.AddInt32(&filteredLen, 1)-1] = g.cachedNodeInfoMap[nodeName].Node()
+				// The scheduler stops searching for more nodes once the remaining nodes number
+				// is less than or equal to 0.
+				atomic.AddInt32(&remainingNodesNumber, -1)
 			} else {
 				predicateResultLock.Lock()
 				failedPredicateMap[nodeName] = failedPredicates
 				predicateResultLock.Unlock()
 			}
 		}
-		numNodesProcessed := int32(0)
-		for numNodesProcessed < allNodes {
-			numNodesToProcess := allNodes - numNodesProcessed
-			if numNodesToProcess > numNodesToFind {
-				numNodesToProcess = numNodesToFind
-			}
-			workqueue.Parallelize(16, int(numNodesToProcess), checkNode)
-			if filteredLen >= numNodesToFind {
-				break
-			}
-			numNodesProcessed += numNodesToProcess
-		}
+
+		// Stops searching for more nodes once the configured number of feasible nodes are found(
+		// once the remainingNodesNumber is less than or equal to 0).
+		workqueue.ParallelizeUntil(16, int(allNodes), checkNode, condition)
```

Comment:
I have a question and need your guidance. If pass the channel instead of synchronization as you said, is it also needed for errs and failedPredicateMap?

Reply:
And the size of the channel buffer is also a problem that needs to be determined.